test/comparePayloads.spec.js

```diff
@@ -0,0 +1,284 @@
+import { it } from "node:test";
+import assert from "node:assert";
+import { join, dirname } from "node:path";
+import { fileURLToPath } from "node:url";
+import { readFileSync } from "node:fs";
+
+// Require Internal Dependencies
+import { comparePayloads } from "../index.js";
+
+// CONSTANTS
+const __dirname = dirname(fileURLToPath(import.meta.url));
+const FIXTURE_PATH = join(__dirname, "fixtures", "scannerPayloads");
+const payload = JSON.parse(readFileSync(join(FIXTURE_PATH, "/payload.json"), "utf8"));
+
+it("should throw an error if compared payloads have the same id", () => {
+  assert.throws(
+    () => compareTo("sameIdPayload"),
+    { message: `You try to compare two payloads with the same id '${payload.id}'` }
+  );
+});
+
+it("should throw an error if compared payloads are not from the same package", () => {
+  assert.throws(
+    () => compareTo("otherRootDependency"),
+    { message: "You can't compare different package payloads 'is-wsl' and 'is-not-wsl'" }
+  );
+});
+
+it("should detect warnings diff", () => {
+  const { warnings: { added, removed }, dependencies: { compared } } = compareTo("warningChangedPayload");
+
+  assert.strictEqual(added.length, 1);
+  assert.deepStrictEqual(added[0], "encoded-literal");
+
+  assert.strictEqual(removed.length, 1);
+  assert.deepStrictEqual(removed[0], "unsafe-regex");
+
+  const deepWarnings = compared.get("foo").versions.compared.get("2.0.0").warnings;
+  assert.strictEqual(deepWarnings.added.length, 1);
+  assert.deepStrictEqual(deepWarnings.added[0], {
+    kind: "unsafe-import",
+    location: [[4, 26], [4, 65]],
+    source: "JS-X-Ray",
+    i18n: "sast_warnings.unsafe_import",
+    severity: "Warning",
+    file: "examples/asyncawait.js"
+  });
+
+  assert.strictEqual(deepWarnings.removed.length, 1);
+  assert.deepStrictEqual(deepWarnings.removed[0], {
+    kind: "unsafe-regex",
+    location: [[3, 16], [3, 55]],
+    source: "JS-X-Ray",
+    i18n: "sast_warnings.unsafe_import",
+    severity: "Warning",
+    file: "examples/asyncawait.js"
+  });
+});
+
+it("should detect flagged authors diff", () => {
+  const { flaggedAuthors: { added, removed } } = compareTo("flaggedAuthorsChanged");
+
+  assert.strictEqual(added.length, 1);
+  assert.deepStrictEqual(added[0], {
+    name: "hugo",
+    email: "hugo@gmail.com"
+  });
+
+  assert.strictEqual(removed.length, 1);
+  assert.deepStrictEqual(removed[0], {
+    name: "jack",
+    email: "jack@gmail.com"
+  });
+});
+
+it("should detect scanner version diff", () => {
+  const { scannerVersion: { prev, now } } = compareTo("scannerVersionChanged");
+
+  assert.strictEqual(prev, "1.0.0");
+  assert.strictEqual(now, "1.0.1");
+});
+
+it("should detect vulnerability strategy version diff", () => {
+  const { vulnerabilityStrategy: { prev, now } } = compareTo("vulnerabilityStrategyChanged");
+
+  assert.strictEqual(prev, "npm");
+  assert.strictEqual(now, "snyk");
+});
+
+it("should detect deep dependencies diff", () => {
```

Comment:
This test has.. Lots of assertions 😅 

I wonder if we should split into multiple tests with multiples (and smallers) payloads

Reply:
I've divided it into several test cases, but I don't know if there's any point in creating several payloads, as there would be a lot of repetition given the large number of mandatory properties.

Reply:
Let me know wyt

Reply:
I see, lgtm 👍